hal/src/argon/platform_ncp_argon.cpp

```diff
@@ -117,12 +117,33 @@ int platform_ncp_fetch_module_info(hal_system_info_t* sys_info, bool create) {
                 info->platform_id = PLATFORM_ID;
                 info->module_function = MODULE_FUNCTION_NCP_FIRMWARE;
 
-                module->info = info;
                 // assume all checks pass since it was validated when being flashed to the NCP
                 module->validity_result = module->validity_checked;
+
+                // IMPORTANT: a valid suffix with SHA is required for the communication layer to detect a change
+                // in the SYSTEM DESCRIBE state and send a HELLO after the NCP update to
+                // cause the DS to request new DESCRIBE info
+                auto suffix = new module_info_suffix_t();
+                if (!suffix) {
+                    delete info;
+                    return SYSTEM_ERROR_NO_MEMORY;
+                }
+                memset(suffix, 0, sizeof(module_info_suffix_t));
+
+                // FIXME: NCP firmware should return some kind of a unique string/hash
+                // For now we simply fill the SHA field with version
+                for (uint16_t* sha = (uint16_t*)suffix->sha;
+                        sha < (uint16_t*)(suffix->sha + sizeof(suffix->sha));
+                        ++sha) {
+                    *sha = version;
+                }
+
+                module->info = info;
```

Comment:
Are the `info` and `suffix` supposed to be released somewhere?

Reply:
Yes, in this function in the `else` condition.

Reply:
So this function is supposed to be called again later on to free the memory by passing `false` as the `create` argument. If so, why don't we implement a separated API to explicitly delete the module info. Or at least add a note that this memory will be freed wherever the module info is no longer used.

Reply:
While I agree that this could be refactored and documented to make things clearer, this is not new functionality and follows the same pattern as https://github.com/particle-iot/device-os/blob/develop/hal/src/nRF52840/ota_flash_hal.cpp#L155 (which is somewhat documented https://github.com/particle-iot/device-os/blob/develop/hal/inc/ota_flash_hal.h#L96) and is not the main purpose of this PR.